lua/entities/gmod_wire_expression2/core/selfaware.lua

```diff
@@ -106,12 +106,35 @@ e2function void setName( string name )
 	e:SetOverlayText(name)
 end
 
--- Get the name of another E2
+-- Set the name of a entity (component name if not E2). Thanks CaptainPRICE for idea and basic implementation
+e2function void entity:setName( string name )
+	if not IsValid(this) or E2Lib.getOwner(self, this) ~= self.player then return end
+	if this:GetClass() == "gmod_wire_expression2" then
+		if this.name == name then return end
+		if name == "generic" or name == "" then
+			name = "generic"
+			this.WireDebugName = "Expression 2"
+		else
+			this.WireDebugName = "E2 - " .. name
+		end
+		this.name = name
+		this:SetNWString( "name", this.name )
+		this:SetOverlayText(name)
+	else
+		if this.wireName == name or string.find(name, "[\n\r\"]") ~= nil then return end
```

Comment:
> Examples of behavior that contributes to creating a positive environment include:
> * Using welcoming and inclusive language
> * Being respectful of differing viewpoints and experiences
> * Gracefully accepting constructive criticism
> * Focusing on what is best for the community
> * Showing empathy towards other community members
> — https://github.com/wiremod/wire/blob/master/CONTRIBUTING.md

Please don't tell others specifically *not* to look at somebody's pull request because it implements similar functionality to another pull request.